### PR TITLE
chore: prevent profiler from being used with versions of node 10 with GC pauses

### PR DESCRIPTION
**Pull request by nolanmar511**

### CODE REVIEW

**Comment by nolanmar511** [comment]
I updated the docs here, and added a link to the github doc indicating which versions of node are allowable.
I also included the issues impacting the versions of node which give us problems. 


Changes to cloud.google.com docs are in flight.

**Comment by aalexand** [comment]
@nolanmar511 Looks like the AppVeyor CI is broken here?

**Comment by nolanmar511** [comment]
Appveyor is still using an older version of node (10.3.0).
I'm re-triggering to see if appveyor is now using a more recent version.

**Comment by nolanmar511** [comment]
PTAL -- tests pass now that AppVeyor uses 10.6.0.

File: README.md [hunk 1]
```diff
@@ -14,8 +14,21 @@ deprecation policy.*
...
+profiler is not stable with those versions of Node.js.
+    * Versions of Node.js 6 prior to 6.12.3 are impacted by
+    [this](https://bugs.chromium.org/p/v8/issues/detail?id=4959) issue, which can
+    cause segmentation faults when heap profiling is enabled.
+    * Versions of Node.js before Node.js 8 prior to 8.9.4 are impacted by
+    [this](https://bugs.chromium.org/p/v8/issues/detail?id=6623) issue, which
+    causes a memory leak when time profiling is enabled.
+    * Versions of Node.js 10 prior to 10.4.1 are impacted by
+    [this](https://bugs.chromium.org/p/chromium/issues/detail?id=847863) issue,
+    which can cause garbage collection to take several minutes when time profiling
```
Comment (aalexand) [inline README.md]: I think that bug is relevant to heap profiling, not time profiling?

Comment (nolanmar511) [reply README.md]: Fixed.

**Comment by nolanmar511** [comment]
PTAL

File: package.json [hunk 2]
```diff
@@ -82,6 +82,6 @@
     ]
   },
   "engines": {
-    "node": "6.12.3 - 7.0.0 || >=8.9.4"
+    "node": ">=6.12.3 <8.0.0 || >=8.9.4 <10.0.0 || >=10.4.1"
```
Comment (aalexand) [inline package.json]: It might be good to add some tests to test this expression.

Comment (nolanmar511) [reply package.json]: Done.

**Pull request merged**